File: tmdb-spring-boot-starter/src/test/java/dev/reuss/tmdb/spring/boot/TmdbAutoConfigurationImportsTest.java
package dev.reuss.tmdb.spring.boot;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

import static org.assertj.core.api.Assertions.assertThat;

class TmdbAutoConfigurationImportsTest {

    private static final String IMPORTS_FILE =
            "META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports";

    @Test
    void autoConfigurationImportsFileContainsTmdbAutoConfigurations() throws IOException {
        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();

        try (var inputStream = classLoader.getResourceAsStream(IMPORTS_FILE)) {
            assertThat(inputStream)
                    .as("AutoConfiguration imports file")
                    .isNotNull();

            String content = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);

            assertThat(content)
                    .contains(TmdbMetricsAutoConfiguration.class.getName())
                    .contains(TmdbClientAutoConfiguration.class.getName())
                    .contains(TmdbServiceAutoConfiguration.class.getName())
                    .contains(TmdbHealthAutoConfiguration.class.getName());
        }
    }
}

File: tmdb-spring-boot-starter/src/test/kotlin/dev/reuss/tmdb/spring/boot/TmdbAutoConfigurationImportsTest.kt
package dev.reuss.tmdb.spring.boot

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class TmdbAutoConfigurationImportsTest {

    @Test
    fun autoConfigurationImportsFileContainsTmdbAutoConfigurations() {
        val classLoader = Thread.currentThread().contextClassLoader

        classLoader.getResourceAsStream(IMPORTS_FILE).use { inputStream ->
            assertThat(inputStream)
                .`as`("AutoConfiguration imports file")
                .isNotNull()

            val content = inputStream
                ?.readAllBytes()
                ?.toString(Charsets.UTF_8)

            assertThat(content)
                .contains(TmdbMetricsAutoConfiguration::class.java.name)
                .contains(TmdbClientAutoConfiguration::class.java.name)
                .contains(TmdbServiceAutoConfiguration::class.java.name)
                .contains(TmdbHealthAutoConfiguration::class.java.name)
        }
    }

    companion object {
        private const val IMPORTS_FILE =
            "META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports"
    }
}